File: java_integration/app/src/main/java/com/firewok/example/java_integration/share/ShareLinkActivity.java
package com.firewok.example.java_integration.share;

import android.net.Uri;
import android.os.Bundle;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;
import com.firework.sdk.FireworkSdk;
import com.firework.sdk.PlayerLaunchResult;
import com.firework.viewoptions.PlayerOption;
import com.firework.viewoptions.ViewOptions;

public class ShareLinkActivity extends AppCompatActivity {
    
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        ViewOptions viewOptions = new ViewOptions.Builder()
                .playerOption(new PlayerOption.Builder()
                        .shareBaseUrl(Constants.BASE_SHARE_URL)
                        .build())
                .build();

        Uri data = getIntent().getData();
        if (data != null) {
            String url = data.toString();
            PlayerLaunchResult playerLaunchResult = FireworkSdk.INSTANCE.startPlayer(viewOptions, url);

            if (playerLaunchResult instanceof PlayerLaunchResult.Failure) {
                PlayerLaunchResult.Failure failure = (PlayerLaunchResult.Failure) playerLaunchResult;
                Toast.makeText(this, failure.getMessage(), Toast.LENGTH_SHORT).show();
            }
        }

        finish();
    }
} 

File: share_link/app/src/main/java/com/firework/example/sharelink/ShareLinkActivity.kt
package com.firework.example.sharelink

import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.firework.example.sharelink.Constants.BASE_SHARE_URL
import com.firework.sdk.FireworkSdk
import com.firework.sdk.PlayerLaunchResult
import com.firework.viewoptions.playerOptions
import com.firework.viewoptions.viewOptions

class ShareLinkActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val viewOptions =
            viewOptions {
                playerOptions {
                    shareBaseUrl(BASE_SHARE_URL)
                }
            }

        val url = intent.data?.toString()!!
        val playerLaunchResult = FireworkSdk.startPlayer(viewOptions = viewOptions, url = url)

        if (playerLaunchResult is PlayerLaunchResult.Failure) {
            Toast.makeText(this, playerLaunchResult.message, Toast.LENGTH_SHORT).show()
        }

        finish()
    }
}
